core/src/forest/ForestPredictor.h

```diff
@@ -53,6 +53,8 @@ class ForestPredictor {
   std::vector<Prediction> predict_oob(const Forest& forest,
                                       const Data& data,
                                       bool estimate_variance) const;
+
+  uint num_threads;
```

Comment:
Instead of storing this variable here, could we pass `num_threads` through to the `PredictionCollector` constructor as we do for `TreeTraverser`?